File: Data Structures & Algorithms/climbing-stairs/submission-1.java
class Solution {
    public int climbStairs(int n) {

        return dp(n) ;

        
        
    }

    private int dp(int n ){
        if(n<=0){
            return 0;
        }
        else if (n==1){
            return 1;
        }
        else if (n==2){
            return 2;
        }
        return dp(n-2)+dp(n-1);
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-1.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        n=len(matrix)
        m=len(matrix[0])
        left=0
        right = n*m-1
        while left<right :
            mid= left + (right-left)//2
            if target<=matrix[mid//n][mid%n] :
                right= mid
            else :
                left=mid+1
        print(left)
        return True if matrix[left//n][left%n]==target else False
        